core/src/main/cpp/gdn/projection_mesh/projection_mesh.h

```diff
@@ -0,0 +1,209 @@
+#ifndef CORE_SRC_MAIN_CPP_GDN_PROJECTION_MESH_PROJECTION_MESH_H_
+#define CORE_SRC_MAIN_CPP_GDN_PROJECTION_MESH_PROJECTION_MESH_H_
+
+#include <core/Ref.hpp>
+#include <core/Vector2.hpp>
+#include <gen/ExternalTexture.hpp>
+#include <gen/MeshInstance.hpp>
+#include <gen/Resource.hpp>
+#include <gen/Shader.hpp>
+#include <gen/ShaderMaterial.hpp>
+#include <gen/Shape.hpp>
+
+#include "projection_mesh_utils.h"
+#include "utils.h"
+
+namespace gast {
+
+namespace {
+using namespace godot;
+const char *kGastTextureParamName = "gast_texture";
+const char *kGastEnableBillBoardParamName = "enable_billboard";
+const char *kGastGradientHeightRatioParamName = "gradient_height_ratio";
+const char *kGastNodeAlphaParamName = "node_alpha";
+const int kDefaultSurfaceIndex = 0;
+const Vector2 kInvalidCoordinate = Vector2(-1, -1);
+}
+
+class ProjectionMesh : Resource {
+GODOT_CLASS(ProjectionMesh, Resource)
```

Comment:
Follow this pattern for the derived `ProjectionMesh` classes.

Reply:
Done.